src/axom/quest/examples/quest_signed_distance_interface.cpp

```diff
@@ -52,21 +55,74 @@ int numranks;
 /*!
  * \brief Holds command-line arguments
  */
-static struct
+struct Arguments
 {
   std::string fileName;
-  int ndims;
-  int maxLevels;
-  int maxOccupancy;
-  int box_dims[3];
-  double box_min[3];
-  double box_max[3];
-  bool specified_box_min;
-  bool specified_box_max;
-  bool is_water_tight;
-  bool dump_vtk;
-  bool use_shared;
-} Arguments;
+  int ndims{3};
+  int maxLevels{15};
+  int maxOccupancy{5};
+  std::vector<int> box_dims{32,32,32};
+  std::vector<double> box_min;
+  std::vector<double> box_max;
+  bool is_water_tight{true};
+  bool dump_vtk{true};
+  bool use_shared{false};
+  bool use_batched_query{false};
+  bool ignore_signs{false};
+
+  void parse( int argc, char** argv, CLI::App& app)
+  {
+    app.add_option("-f,--file", this->fileName, "specifies the input mesh file")
+      ->check(CLI::ExistingFile)
+      ->required();
+
+    app.add_option("--dimension", this->ndims, "the problem dimension")
+      ->capture_default_str();
+
+    app.add_option("--maxLevels", this->maxLevels, "max levels of Subdivision for the BVH")
+      ->capture_default_str();
+
+    app.add_option("--maxOccupancy", this->maxOccupancy, "max number of item per BVH bin")
+      ->capture_default_str();
+
+    app.add_option("--box-dims", box_dims, "the dimensions of the box mesh")
+      ->expected(3);
+
+    auto* minbb = app.add_option("--box-min", box_min, "the lower corner of the box mesh")
+      ->expected(3);
+    auto* maxbb = app.add_option("--box-max", box_max, "the upper corner of the box mesh")
+      ->expected(3);
+    minbb->needs(maxbb);
+    maxbb->needs(minbb);
+
+    app.add_flag("!--no-vtk", this->dump_vtk, "disables VTK output")
+      ->capture_default_str();
+
+    app.add_flag("!--not-watertight", this->is_water_tight, "indicates that input is not water-tight")
+      ->capture_default_str();
+
+    app.add_flag("--use-shared", this->use_shared, "stores the surface using MPI-3 shared memory")
+      ->capture_default_str();
+
+    app.add_flag("--batched", this->use_batched_query, 
+      "uses a vectorized query instead of many serial queries")
+      ->capture_default_str();
+
+    app.add_flag("--ignore-signs", this->ignore_signs, 
+      "distance query should ignore signs")
+      ->capture_default_str();
+
+    app.get_formatter()->column_width(40);
+
+    // could throw an exception
+    app.parse( argc, argv);
```

Comment:
How does `CLI` handle the `--help`, does it handle that internally?

Reply:
Yes, It is automatic,

Reply:
For convenience, here is the output:
```
>./examples/quest_signed_distance_interface_ex -h
Driver for signed distance query
Usage: ./examples/quest_signed_distance_interface_ex [OPTIONS]

Options:
  -h,--help                             Print this help message and exit
  -f,--file TEXT:FILE REQUIRED          specifies the input mesh file
  --dimension INT=3                     the problem dimension
  --maxLevels INT=15                    max levels of subdivision for the BVH
  --maxOccupancy INT=5                  max number of item per BVH bin
  --box-dims INT x 3                    the dimensions of the box mesh
  --box-min FLOAT x 3 Needs: --box-max  the lower corner of the box mesh
  --box-max FLOAT x 3 Needs: --box-min  the upper corner of the box mesh
  --no-vtk{false}                       disables VTK output
  --not-watertight{false}               indicates that input is not water-tight
  --use-shared                          stores the surface using MPI-3 shared memory
  --batched                             uses a single batched query on all points instead of many individual queries
  --ignore-signs                        distance query should ignore signs

```

Reply:
Nice!